apps/bbsync/query.py

```diff
@@ -157,6 +157,17 @@ def generate_flags(self):
     ]
     EMBARGOED_GROUPS = ["qe_staff", "security"]
 
+    def _standardize_embargoed_groups(self, groups):
+        """
+        combine groups with default embargoed
+        but make sure all of them are allowed
+        plus always remove redhat group
+        """
+        return list(
+            ((set(groups) | set(self.EMBARGOED_GROUPS)) & set(self.ALLOWED_GROUPS))
+            - set("redhat")
```

Comment:
if I understand the goal of the function correctly it is to:

1. guarantee that `EMBARGO_GROUPS` are always present, i.e. it's the base
2. that the groups passed in belong to `ALLOWED_GROUPS` because it's possible that e.g. jira groups or some other random groups we don't care about are being passed in
3. to ignore the `"redhat"` group (because it's implicit?)

Thus it might be better to write it like so:

`set(self.EMBARGOED_GROUPS) | (set(groups) & set(self.ALLOWED_GROUPS)) - {"redhat"}`

Reply:
You understand it correctly. I not entirely sure about all the reasons but I think it is some kind of safeguard. The Jira is not the issue. The issue might be if someone fills the wrong groups in the product definitions so I think the safeguard is (or was) preventing our people mistakes. I can definitely restructure it your way.